src/main/java/org/opensearch/ad/indices/AnomalyDetectionIndices.java

```diff
@@ -406,7 +406,22 @@ public void initAnomalyDetectorJobIndex(ActionListener<CreateIndexResponse> acti
         try {
             CreateIndexRequest request = new CreateIndexRequest(AnomalyDetectorJob.ANOMALY_DETECTOR_JOB_INDEX)
                 .mapping(AnomalyDetector.TYPE, getAnomalyDetectorJobMappings(), XContentType.JSON);
-            choosePrimaryShards(request);
+            request
+                .settings(
+                    Settings
+                        .builder()
+                        // AD job index is small. 1 primary shard is enough
+                        .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
```

Comment:
Should we or is it possible to change the old AD job index setting to this new setting?

Reply:
You meant changing the number of primary shards? if yes, OpenSearch does not support it.

Reply:
Oh, I mean the auto expand replica, so we can leverage more data nodes to run realtime job.

Reply:
changed.